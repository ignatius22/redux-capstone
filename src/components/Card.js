const Card = ({ title, image, calories }) => {
  return (
    <div data-testid='card' className="tc grow bg-navy br3 pa2 ma3 dib bw2 shadow-3 ">
      <img alt="recipes" src={image} />
      <div>
        <h2>{title}</h2>
        <p>{calories}</p>
      </div>
    </div>
  );
};

export default Card;