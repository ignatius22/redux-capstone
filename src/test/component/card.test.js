/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest';
import Card from '../../components/Card';
import ReactDOM from 'react-dom'


afterEach(cleanup);

it('* renders *', () => {
  const { asFragment } = render(<Card label='any props'/>);
  expect(asFragment()).toMatchSnapshot();
});

it('* renders without crashing *', () => {
	const div = document.createElement('div')
	ReactDOM.render(<Card />, div)
})
