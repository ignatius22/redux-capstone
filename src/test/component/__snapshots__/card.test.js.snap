// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`* renders * 1`] = `
<DocumentFragment>
  <div
    class="tc grow bg-navy br3 pa2 ma3 dib bw2 shadow-3 "
    data-testid="card"
  >
    <img
      alt="recipes"
    />
    <div>
      <h2 />
      <p />
    </div>
  </div>
</DocumentFragment>
`;
